projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php

```diff
@@ -179,13 +179,21 @@ public function create_item( $request ) {
 		$plugins  = $request['plugins'];
 		usort( $plugins, 'strnatcasecmp' );
 
-		$event = Scheduled_Updates::create_scheduled_update( $schedule['timestamp'], $schedule['interval'], $plugins );
+		$event = wp_schedule_event( $schedule['timestamp'], $schedule['interval'], Scheduled_Updates::PLUGIN_CRON_HOOK, $plugins, true );
 
 		if ( is_wp_error( $event ) ) {
+			// If the schedule could not be created, return an error.
+			$event->add_data( array( 'status' => 500 ) );
+
 			return $event;
 		}
 
-		$id = Scheduled_Updates::generate_schedule_id( $plugins );
+		$id    = Scheduled_Updates::generate_schedule_id( $plugins );
+		$event = wp_get_scheduled_event( Scheduled_Updates::PLUGIN_CRON_HOOK, $plugins, $schedule['timestamp'] );
+
+		if ( ! $event ) {
+			return new WP_Error( 'rest_invalid_schedule', __( 'The schedule could not be created.', 'jetpack-scheduled-updates' ), array( 'status' => 500 ) );
```

Comment:
How about 422? (Unprocessable Content)

Reply:
> Warning: The client should not repeat this request without modification.
> —https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

I think this is just a really tricky situation

Reply:
I've put on 500 because this shouldn't (but unfortunately do) happen when `wp_schedule_event` don't send back a `WP_Error`. You see it in other failing functions like `wp_update_comment`.

A 404 is better here because something has been created but cannot be found later?

Reply:
> I think this is just a really tricky situation

Yeah, that's true. In this case, 404 looks more suitable for our situation here.

Reply:
Changed to 404, thanks.